packages/cozy-stack-client/src/CozyStackClient.js

```diff
@@ -191,44 +191,36 @@ class CozyStackClient {
       credentials: 'include'
     }
 
-    if (!global.document) {
-      throw new Error('Not in a web context, cannot refresh token')
-    }
-
     const response = await fetch('/?refreshToken', options)
 
     if (!response.ok) {
       throw new Error(
-        "couldn't fetch a new token - response " + response.statusCode
+        "Couldn't fetch a new token - response " + response.statusCode
       )
     }
     const html = await response.text()
-    const parser = new DOMParser()
-    const doc = parser.parseFromString(html, 'text/html')
-    if (!doc) {
-      throw Error("couldn't fetch a new token - doc is not html")
-    }
-    const appNode = doc.querySelector('div[role="application"]')
-    if (!appNode) {
-      throw Error("couldn't fetch a new token - no div[role=application]")
+    const match = html.match(/data-cozy="([^"]+)"/)
+    if (!match || match.length < 2) {
+      throw new Error(
+        "Couldn't fetch a new token - no data-cozy element found in HTML"
+      )
     }
-    const data = appNode.dataset.cozy
-      ? JSON.parse(appNode.dataset.cozy)
-      : { ...appNode.dataset }
+    const htmlCozyData = match[1].replace(/&#34;/g, '"')
+    const cozyData = JSON.parse(htmlCozyData)
 
-    let { token } = data
+    let { token } = cozyData
     if (token) {
-      token = token || data.cozyToken
+      token = token || cozyData.cozyToken
     }
```

Comment:
This block can be removed, `data.cozyToken` is from the `appNode.dataset` you removed.

Reply:
Here is the result when calling `document.querySelector('div[role="application"]').dataset`:
![image](https://github.com/user-attachments/assets/1d4e7797-c348-4cfc-9792-ba8a07fa5282)

The `dataset.cozy` value comes from the `data-cozy` attribute

All the `dataset.cozyXxx` values come from the deprecated `data-cozy-xxx`